addon/lib/infinity-model.js

```diff
@@ -179,5 +180,50 @@ export default ArrayProxy.extend({
     }
 
     return objectAssign(pageParams, get(this, 'extraParams'));
+  },
+
+  /**
+    orchestrate after-model hooks, local to the class, or configured on the route
+
+    @private
+    @method _afterInfinityModel
+    @return {Ember.RSVP.Promise} A Promise that resolves the new objects
+    @return {Ember.Array} the new objects
+   */
+  _afterInfinityModel(newObjects, infinityModel) {
+    let result = this.afterInfinityModel(newObjects, infinityModel);
+
+    if(!this._routeAfterInfinityModel || typeof this._routeAfterInfinityModel !== 'function') {
+      return result || newObjects;
+    } else {
+      return result.then((result) => { return this._routeAfterInfinityModel(result || newObjects, infinityModel) || newObjects });
```

Comment:
if it fails the first check (`this._routeAfterInfinityModel`), then this still could fail if it didn't exist.